nvflare/job_config/fed_job.py

```diff
@@ -349,6 +383,91 @@ def _add_referenced_components(self, base_component, target):
                             # remove already added components from tracked components
                             self._components.pop(base_id)
 
+    def _add_script_executor(self, obj, target, tasks):
```

Comment:
This logic doesn't belong here! You can define a method like add_to_job() for JobObj class, and let the ScriptExecutorJobObj implement this detail in its own way.

The fed_job module is framework level code. It should be coded based on abstraction. If what we defined for JobObj is not enough, you can expand its abstraction.